cabal-install/Distribution/Client/ProjectConfig/Types.hs

```diff
@@ -261,7 +261,8 @@ data PackageConfig
        packageConfigHaddockCss          :: Flag FilePath, --TODO: [required eventually] use this
        packageConfigHaddockHscolour     :: Flag Bool, --TODO: [required eventually] use this
        packageConfigHaddockHscolourCss  :: Flag FilePath, --TODO: [required eventually] use this
-       packageConfigHaddockContents     :: Flag PathTemplate --TODO: [required eventually] use this
+       packageConfigHaddockContents     :: Flag PathTemplate, --TODO: [required eventually] use this
+       packageConfigHaddockForHackage   :: Flag HaddockTarget
```

Comment:
Fwiw, this looks somewhat inconsistent; we got a field named `ForHackage` which would imply a Bool, but then this holds an enum which redundantly says `ForHackage` (or `ForDevelopment`).